tests/caasp/health_check.pm

```diff
@@ -0,0 +1,90 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Check that health-check service works correctly
+# Maintainer: Ciprian Cret <ccret@suse.com>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use caasp;
+use utils;
+use power_action_utils qw(power_action);
+
+
+sub get_btrfsid {
+    my $btrfs_id = script_output("btrfs subvolume get-default /");
+    $btrfs_id =~ /ID (\d+) gen/;
+    $btrfs_id = $1;
+    return $btrfs_id;
+}
+
+sub get_loggedid {
+    my $logged_id = script_output("cat /var/lib/misc/health-check.state");
+    $logged_id =~ /LAST_WORKING_BTRFS_ID=(\d+)/;
+    $logged_id = $1;
+    return $logged_id;
+}
+
+sub compare_id {
+    my $btrfs_id = get_btrfsid;
+    my $hc_id    = get_loggedid;
+    die "The current snapshot id does not match the one from the health-checker log" unless $hc_id == $btrfs_id;
+}
+
+sub run {
+    # run health-checker to make sure the current snapshot doesn't have issues
+    validate_script_output("health-checker", sub { m/passed/ });
+
+    # keep the current snapshot id and compare it to the logged one
+    my $initial_id = get_btrfsid;
+    compare_id;
+
+    # update etcd.sh to force health-checker to fail
+    type_string("transactional-update shell\n");
+    type_string("cp /usr/lib/health-checker/etcd.sh /tmp/etcd_bk.sh\n");
+    type_string("sed -i 's/exit 0/exit 1/g' /usr/lib/health-checker/etcd.sh\n");
+    type_string("exit\n");
+
+    # check that the changes applied and we have a new snapshot
+    my $current_id = get_btrfsid;
+    my $logged_id  = get_loggedid;
+    die "The current snapshot is not ahead of the logged one" unless $current_id > $logged_id;
+
+    power_action('reboot');
```

Comment:
you can use `process_reboot 1` sub from caasp - it will reboot & relogin

Reply:
Updated.
New validation run: http://ccret.suse.cz/tests/3138#